### ISSUE TITLE
Coal Synthesizer is Significantly Worse than Coal Centrifuge

### ISSUE THREAD

**Issue by ridesdragons** [genNAowl/Upgraded-Factories#2]
I was redesigning one of my older bases to streamline it for certain production lines and needed to redo my power supply, when I noticed that steam engines could use multiple types of fuel, which got me interested in calculating if there was a better fuel source than coal for them (spoiler alert: there isn't). That's when I noticed an anomaly - the Nurturer is better than the Cultivator, and the Spore Compactor better than the Spore Press, which was to be expected from upgraded structures. I then created a hypothetical fully-closed setup and ran the numbers when I noticed... I was getting less than double the amount of coal out as I was putting spore pods in.

Normally, 1 spore pod becomes 6 oil, when then becomes 2 coal, for a 1 : 6 : 2 ratio, and replacing the Press with the Compactor ups that ratio up to 1 : 7.5 : 2.75. However, if you then replace the Centrifuge with a Synthesizer, this ratio drops down to a 1 : 6.43 : 1.43 ratio. That's just abyssmal.

I stopped and thought "yea but maybe it's more energy efficient, or uses up less space, or something", and decided to run those numbers. for a perfectly closed system, the first step of both setups would have 24 Nurturers and 5 Compactors, which require 13020 power and 261 tiles (excluding pipes and conveyors).

For the Centrifuge-based setup (setup C), the next step would require a whopping 75 centrifuges, consuming 3150 power and taking up 300 tiles (producing 165 coal).

Meanwhile, the Synthesizer-based setup (setup S) would require and additional 4 Nurturers and 10 Synthesizers, consuming 3420 power and taking up 126 tiles (producing 100 coal).

While I was surprised that the setup S was actually _more_ power-hungry than setup C, I thought I had finally gotten my answer in regards to space. Setup C needed 561 tiles of space at the minimum, whereas setup S only required 387, roughly 45% bigger. Considering space is a resource of its own, I thought maybe the drastically reduced fuel efficiency was worth it on saving space. That is... until I did my next calculation.

Generally, the main point of turning water into coal is to feed into a steam engine (and making steam setups was the reason for me calculating this in the first place). So, I took the coal generated from these hypothetical setups, threw them into some steam engines, subtracted the energy costs, and... setup C produced 65505 surplus energy, while setup S only produced 33060 surplus energy, only little more than half. Setup C produces 116.76 energy per tile (or 42.18 when factoring in the steam engine tiles), while setup S produces only 85.4 energy per tile (33.5 when factoring in steam engine tiles). Setup C is 36.7% (25.9%) more energy-dense than setup S, meaning you're not actually saving any space on using synthesizers, either.

Naturally, the surplus energy on these gets worse the more pumps and drills you use instead of mechanical pumps, and additional power expenditures hurt setups with a smaller surplus more than those with a larger setup. It doesn't help that setup S also ends up needing more water than setup C....

In summary, Synthesizers are larger blocks that add complications to their inputs while consuming more power while ultimately producing less coal total and coal per tile than their vanilla counterpart. All down-sides, no up-sides.

**Comment by genNAowl** [comment]
Coal Synthesizer has been buffed: 
- Now uses 33% less oil
- Power cost has almost been reduced by half

I have not made sure that most of the buildings in this mod are balanced, so feedback is appreciated. Thanks.

**Comment by ridesdragons** [comment]
no problem m8, will let you know if I find anything else (so far everything else seems to be fairly strong. some are certainly stronger than others, this was just the only one that I've found so far that was actually weaker than vanilla)

regarding the buff you've given to the coal synthesizer, I've plugged the new numbers into my hypothetical setup, and here are the results:

power consumption increases from 3420 to 4305 (+885, less is better)
tile usage increases from 126 to 189 (+63, less is better)
power surplus increases from 33060 to 56925 (+23865, more is better)
power density increases from 85.4 (33.5) to 126.5 (42.1666) (more is better)

because the oil consumption was reduced from 45/s to 30/s, in order to use the same 450 oil, I'd need to increase the number of synthesizers from 10 to 15, which also tacks on an additional 2 nurturers. honestly, the lion's share of the power inefficiency of this setup is actually from the nurturers, not the synthesizer itself. not saying you should reduce nurturer power consumption, just pointing out where the energy bleed is coming from. however, the additional 50 coal produced from the synthesizers bumped up the energy density drastically, making them roughly equal to centrifuges on power production.

as for the reason the power density is higher before calculating steam engines and lower after, this is due to the ratio of power prep buildings to power production buildings (the setup, which doesn't produce any power, vs the steam engines that only produce power). generally speaking, the higher the ratio, the better, as it means that for every tile used creating coal, you're using R tiles to turn the coal into power. the setup C had a ratio of 1:1.768 (first number divided by second number, minus 1), while pre-buff setup S had a ratio of 1:1.549 and post-buff has a ratio of 1:2. so as far as tile usage is concerned, it's now more tile-efficient that setup C. however, the higher the tile-efficiency, the more it's affected by resource and power efficiency. because of the higher power consumption and slightly lower resource efficiency, it ends up only ranking just above pre-engine, and just below post-engine (and post-engine matters more).

as for resource efficiency, it's still less efficient than setup C. on one hand, reducing oil consumption brought it back down to the vanilla 3 oil -> 1 coal. however, because of the addition of the spore pod, which itself is used to make oil, actual efficiency is still somewhat worse. remember that the vanilla ratio is 1 : 6 : 2. setup C's ratio was 1 : 7.5 : 2.5. setup S pre-buff was 1 :  6.43 : 1.43. post-buff, it is now 1 : 6 : 2. in other words, the efficiency in the 3rd step (synthesizer) is bad enough to undo the efficiency gained in the first 2 steps (nurturer, compactor).

so what you gain in tile-efficiency, you lose in resource efficiency.

I personally would suggest oil consumption be reduced to at most half, at 22.5, instead of 30, though (1 spore pod pretty much equals 7.5 oil, anyway, with the compactor). at 22.5 oil consumption, the ratio would jump up to 1 : 5.625 : 2.5. since that last number is all that really matters, this would mean it'd be as efficient at turning water into coal as setup C. that said, since it's an upgrade, it'd be preferable to have a better ratio than setup C, but at least at that point it'd be more tile efficient while being as resource efficient (and not using more water than the other setup, meaning it's not damned more by maps with no surface water. water drills are very energy hungry lol). since the first two steps bumped resource efficiency from 1 : 2 to 1 : 2.5, maybe synthesizers bumping that further to 2.75 would be balanced. though, at this point, instead of lowering oil consumption even further, I think it'd be better to raise coal output instead. I'm guessing you have it set to 10 in order to have it easily fit on a belt? but, honestly, I'm already used to advanced structures needing more belts to keep up with output, as are other people, I'm sure. hell, the vanilla airblast drill, when water-cooled and covering enough resources, outputs more than a single belt can support. one such drill on a 4x4 square of sand will produce 11.1 sand per second.

overall, I think reducing oil consumption to 22.5 and increasing coal production to 11, or keeping oil at 30 while increasing coal production to 14/15, would provide enough of an efficiency boost in resources to warrant using synthesizers over centrifuges, especially on maps without surface water, without making them overpowered. if you want to double down on the upgrades being more space efficient, the 2nd suggestion is better (tweaking inputs has an effect on energy density, but tweaking outputs has a more pronounced result, as the number of buildings in the setup doesn't increase)

oh god I need to sleep, I've been sitting here for 2 hours crunching numbers instead of sleeping aaa

**Comment by genNAowl** [comment]
Buffed again, from 10 coal output to 15

**Comment by ridesdragons** [comment]
fast lol. I'll let you know when I use the setup how it feels and such. I'll also make sure to inform you if I find anything else that's off. thanks m8

btw, are the oil and water extorters disabled for now? saw them in the files, but couldn't find them in the tech tree, or in sandbox mode

**Comment by genNAowl** [comment]
Yeah they are disabled because: 
1. Their sprites are really bad
2. I don't know if they are really needed in a mod like this
